File: frontend/src/scripts/iot_handler.js
// Used ChatGPT to translate these from Python

import {IoTSimulator} from "./iot_simulator"
import {herbicideFilter, insecticideFilter, fungicideFilter, fertilizerFilter, chemicalSum} from "./iot_filter"

function gen(ct) {
    const dg = new IoTSimulator(); 
    let currHerb = [];
    let herbSpray = false;
    let currInsect = [];
    let insectSpray = false;
    let currFungi = [];
    let fungiSpray = false;
    let currFert = [];
    let useFert = false;

    for (let x = 0; x < ct; x++) {
        let out = [];

        // Herbicide processing
        out.push(dg.timer({ herbicide: true }));
        let curr = herbicideFilter(out[0]);
        if (curr !== null) {
            currHerb.push(curr);
            herbSpray = true;
        } else if (herbSpray) {
            herbSpray = false;
            console.log(chemicalSum(currHerb));
            currHerb = [];
        }
        if (herbSpray && x === ct - 1) {
            herbSpray = false;
            console.log(chemicalSum(currHerb));
            currHerb = [];
        }

        // Insecticide processing
        out.push(dg.timer({ insecticide: true }));
        curr = insecticideFilter(out[1]);
        if (curr !== null) {
            currInsect.push(curr);
            insectSpray = true;
        } else if (insectSpray) {
            insectSpray = false;
            console.log(chemicalSum(currInsect));
            currInsect = [];
        }
        if (insectSpray && x === ct - 1) {
            insectSpray = false;
            console.log(chemicalSum(currInsect));
            currInsect = [];
        }

        // Fungicide processing
        out.push(dg.timer({ fungicide: true }));
        curr = fungicideFilter(out[2]);
        if (curr !== null) {
            currFungi.push(curr);
            fungiSpray = true;
        } else if (fungiSpray) {
            fungiSpray = false;
            console.log(chemicalSum(currFungi));
            currFungi = [];
        }
        if (fungiSpray && x === ct - 1) {
            fungiSpray = false;
            console.log(chemicalSum(currFungi));
            currFungi = [];
        }

        // Fertilizer processing
        out.push(dg.timer({ fertilizer: true }));
        curr = fertilizerFilter(out[3]);
        if (curr !== null) {
            currFert.push(curr);
            useFert = true;
        } else if (useFert) {
            useFert = false;
            console.log(chemicalSum(currFert));
            currFert = [];
        }
        if (useFert && x === ct - 1) {
            useFert = false;
            console.log(chemicalSum(currFert));
            currFert = [];
        }
    }
    console.log("\n")
}

export{
    gen
}

File: ignition/support/old/iot_handler.py
import iot_simulator as sim
import iot_filter as filter



def gen():
    ct = 50

    dg = sim.IoTSimulator()
    curr_herb = []
    herb_spray = False
    curr_insect = []
    insect_spray = False
    curr_fungi = []
    fungi_spray = False
    curr_fert = []
    use_fert = False
    for x in range(ct):
        out = []
        out.append(dg.timer(herbicide=True))
        curr = filter.herbicide_filter(out[0])
        if curr != None:
            print(curr)
            curr_herb.append(curr)
            herb_spray = True
        elif herb_spray:
            herb_spray = False
            print(filter.chemical_sum(curr_herb))
            curr_herb = []
            print("\n")
        if herb_spray and x == ct-1:
            herb_spray = False
            print(filter.chemical_sum(curr_herb))
            curr_herb = []
            print("\n")
    
        out.append(dg.timer(insecticide=True))
        curr = filter.insecticide_filter(out[1])
        if curr != None:
            print(curr)
            curr_insect.append(curr)
            insect_spray = True
        elif insect_spray:
            insect_spray = False
            print(filter.chemical_sum(curr_insect))
            curr_insect = []
            print("\n")
        if insect_spray and x == ct-1:
            insect_spray = False
            print(filter.chemical_sum(curr_insect))
            curr_insect = []
            print("\n")
        
        out.append(dg.timer(fungicide=True))
        curr = filter.fungicide_filter(out[2])
        if curr != None:
            print(curr)
            curr_fungi.append(curr)
            fungi_spray = True
        elif fungi_spray:
            fungi_spray = False
            print(filter.chemical_sum(curr_fungi))
            curr_fungi = []
            print("\n")
        if fungi_spray and x == ct-1:
            fungi_spray = False
            print(filter.chemical_sum(curr_fungi))
            curr_fungi = []
            print("\n")

        out.append(dg.timer(fertilizer=True))
        curr = filter.fertilizer_filter(out[3])
        if curr != None:
            print(curr)
            curr_fert.append(curr)
            use_fert = True
        elif use_fert:
            use_fert = False
            print(filter.chemical_sum(curr_fert))
            curr_fert = []
            print("\n")
        if use_fert and x == ct-1:
            use_fert = False
            print(filter.chemical_sum(curr_fert))
            curr_fert = []
            print("\n")

    # print(dg.used_ids)

gen()